internal/gengapic/client_init.go

```diff
@@ -135,155 +135,236 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
 	return nil
 }
 
-func (g *generator) clientInit(serv *descriptor.ServiceDescriptorProto, servName string) error {
+func (g *generator) abstractClientIntfInit(serv *descriptor.ServiceDescriptorProto, servName string) error {
 	p := g.printf
-
-	var hasLRO bool
+	p("// %sAbstractClient is an interface that defines the methods availaible from %s.", servName, g.apiName)
+	p("// It is agnostic as to the underlying transport, i.e. json+http, gRPC, or other.")
+	p("type %sAbstractClient interface {", servName)
 	for _, m := range serv.Method {
-		if g.isLRO(m) {
-			hasLRO = true
-			break
+		inType := g.descInfo.Type[*m.InputType]
```

Comment:
Nit: Please use the field accessor instead of direct access.
```suggestion
		inType := g.descInfo.Type[m.GetInputType()]
```

Reply:
Done